src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -422,119 +416,28 @@ private static bool FileValidationEnabledForDataType(DataType dataTypeFromMetada
                     return BadRequest($"Could not determine if data type {dataType?.Id} requires application logic.");
                 }
 
-                var modelType = _appModel.GetModelType(dataType.AppLogic.ClassRef);
-
-                var oldModel =
-                    await _dataClient.GetFormData(instanceGuid, modelType, org, app, instanceOwnerPartyId, dataGuid);
+                ServiceResult<DataPatchResult, DataPatchError> res = await _patchService.ApplyPatch(instance, dataType, dataElement, dataPatchRequest.Patch, language, dataPatchRequest.IgnoredValidators);
 
-                var (response, problemDetails) =
-                    await PatchFormDataImplementation(dataType, dataElement, dataPatchRequest, oldModel, language, instance);
-
-                if (problemDetails is not null)
+                if (res.Success)
                 {
-                    return StatusCode(problemDetails.Status ?? 500, problemDetails);
+                    await UpdateDataValuesOnInstance(instance, dataType.Id, res.Ok.NewDataModel);
+                    await UpdatePresentationTextsOnInstance(instance, dataType.Id, res.Ok.NewDataModel);
+                    
+                    return Ok(new DataPatchResponse
+                    {
+                        NewDataModel = res.Ok.NewDataModel,
+                        ValidationIssues = res.Ok.ValidationIssues
+                    });
                 }
 
-                await UpdatePresentationTextsOnInstance(instance, dataType.Id, response.NewDataModel);
-                await UpdateDataValuesOnInstance(instance, dataType.Id, response.NewDataModel);
-
-                // Save Formdata to database
-                await _dataClient.UpdateData(
-                    response.NewDataModel,
-                    instanceGuid,
-                    modelType,
-                    org,
-                    app,
-                    instanceOwnerPartyId,
-                    dataGuid);
-
-                return Ok(response);
+                return Problem(res.Error);
```

Comment:
The function name `Problem` looks too much like the standard `Ok`, `NotFound`, `BadRequest` to have different behavior from them. A more descriptive name would be more appropriate.

Reply:
Ok, Looking at this once more. There is a `Problem` method on the base class, with very similar behaviour, so I retract this comment.